src/peppr/rotamer.py

```diff
@@ -0,0 +1,1439 @@
+
+    This class represents the rotamer score for a single residue, including its name,
+    ID, observed chi angles, rotamer score percentage, and classification.
+
+    Attributes
+    ----------
+    resname : str
+        The name of the residue (e.g. "ARG", "LEU").
+    resid : int
+        The ID of the residue in the structure.
+    chain_id : str
+        The chain ID of the residue.
+    angles : dict[str, float]
+        A dictionary containing the observed chi angles for the residue, e.g. {'chi1': 60.0, 'chi2': -45.0}.
+    pct : float
+        The rotamer score percentage for the residue, e.g. 0.01 for 1% favored.
+    classification : ConformerClass
+        The classification of the residue based on its rotamer score percentage.
+        It can be one of the following:
+        - "FAVORED": The residue is in a favored rotamer conformation.
+        - "ALLOWED": The residue is in an allowed rotamer conformation.
+        - "OUTLIER": The residue is in an outlier rotamer conformation.
+        - "UNKNOWN": The residue's rotamer score could not be determined.
+
+    References
+    ----------
+    .. [1] https://pmc.ncbi.nlm.nih.gov/articles/PMC4983197/
+    """
+
+    res_name: str
+    res_id: int
+    chain_id: str
+    angles: dict[str, float]
+    pct: float
+    classification: ConformerClass
+
+    @classmethod
+    def from_residue(
+        cls, atom_array: struc.AtomArray, res_id: int, res_name: str, chain_id: str
+    ) -> "ResidueRotamerScore":
+        """
+        Create ResidueRotamerScore from a residue object.
+
+        Parameters
+        ----------
+        atom_array : struc.AtomArray
+            The structure.
+        res_id : int
+            The residue ID to check.
+        res_name : str
+            The residue name to check (e.g. "ARG", "LEU").
+        chain_id : str
+            The chain ID to check.
+
+        Returns
+        -------
+        ResidueRotamerScore
+            An instance of ResidueRotamerScore containing the rotamer score for the residue.
+        """
+        result = _check_rotamer(atom_array, res_id, chain_id)
+        score, angles = result
+        chis = {f"chi{i + 1}": ang for i, ang in enumerate(angles)}
+
+        return cls(
+            res_name=res_name,
+            res_id=res_id,
+            chain_id=chain_id,
+            angles=chis,
+            **asdict(score),
+        )
+
+
+def _classify_rotamers(atom_array: struc.AtomArray) -> list["ResidueRotamerScore"]:
+    """
+    Create ResidueRotamerScore from a protein structure.
+
+    Parameters
+    ----------
+    atom_array : struc.AtomArray
+        The structure.
+
+    Returns
+    -------
+    list[ResidueRotamerScore]
+        An instance of RotamerScore containing the rotamer scores for the structure.
+    """
+    if not isinstance(atom_array, struc.AtomArray):
+        warnings.warn("RotamerScore: Cannot be computed; atom_array must be AtomArray")
+        return []
```

Comment:
This would clearly indicate a user error, right? In this case a `TypeError` would be more appropriate.

In general, I would discourage such type checks, if they do not target very common pitfalls, as they quite bloat the code, if done thoroughly.